Firestore/core/src/firebase/firestore/remote/grpc_connection.mm

```diff
@@ -210,30 +226,41 @@
 }
 
 /*static*/ void GrpcConnection::UseTestCertificate(
-    absl::string_view certificate_path, absl::string_view target_name) {
-  HARD_ASSERT(!certificate_path.empty(), "Empty path to test certificate");
+    const std::string& host,
+    const Path& certificate_path,
+    const std::string& target_name) {
+  HARD_ASSERT(!host.empty(), "Empty host name");
+  HARD_ASSERT(!certificate_path.native_value().empty(),
+              "Empty path to test certificate");
   HARD_ASSERT(!target_name.empty(), "Empty SSL target name");
 
-  if (!test_credentials_) {
+  if (!config_by_host_) {
     // Deliberately never deleted.
-    test_credentials_ = new TestCredentials{};
+    config_by_host_ = new ConfigByHost{};
```

Comment:
Done (using the second approach).

The reason I did it that way is because of the style guide. However, in this case I'm quite confident `config_by_host` won't be accessed after it's destructed (which would be a more serious bug anyway).